### PR TITLE
Index work items upon payment request

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- bootstrap rocket server
- move variable where it is used
- new index repo route
- restore deleted functions
- create issue indexing route
- create pr indexing route
- create user indexing route
- do not save state if nothing has been indexed
- remove unused generic parameter
- new indexer service
- mount all api routes under the same base
- move request and cancel payment mutation to REST api
- extract event storing in it
- new it for payment request
- new it for access control on payment request
- new it for payment cancel
- rename tests
- admin can cancel a payment
- use user github access token
- make payment a data struct
- remove unused usecases in graphql context
- add work item type
- create http client from request to forward authorization header
- index work items upon payment request
- add indexer client config
- add it for indexer calls

### CODE REVIEW

File: backend/api/src/bin/events_sanity_checks/app.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 default:
   http:
     api_keys:
-      graphql: $BACKEND_GRAPHQL_API_KEY
```
Comment (PierreOucif) [inline backend/api/src/bin/events_sanity_checks/app.yaml:4]: J'ai pas compris ce changement ?

**Review by ofux — APPROVED** [review]

File: backend/common/infrastructure/src/github/clients/mod.rs [hunk 2]
```diff
@@ -221,6 +221,24 @@ impl Client {
...
+		issue_number: GithubIssueNumber,
+	) -> Result<Issue, Error> {
+		let issue: Issue = self
+			.get_as(format!(
+				"{}repositories/{repo_id}/issues/{issue_number}",
+				self.octocrab().base_url
+			))
+			.await?;
+		match issue.pull_request {
+			Some(_) => Err(Error::NotFound(anyhow!("Issue is in fact a pull request"))),
```
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply backend/common/infrastructure/src/github/clients/mod.rs:237]: oui si on appelle l'endpoint avec un pr number
genre: `https://api.github.com/repos/onlydustxyz/marketplace/issues/1140`
On decide ici de rejeter le call de notre cote contrairement a github
ca permet de s'assurer qu'on indexe avec les bons ids

File: backend/api/src/bin/events_sanity_checks/app.yaml (hunk 1 above)
Comment (AnthonyBuisset) [reply backend/api/src/bin/events_sanity_checks/app.yaml:4]: on passe de `Map` a `Vec`

File: backend/common/presentation/src/http/guards/api_key.rs [hunk 3]
```diff
@@ -20,30 +14,26 @@ pub enum ApiKeyError {
...
-			.rocket()
-			.state::<Config>()
-			.and_then(|config| config.api_keys().get(K::name()))
+		if let Some(allowed_keys) =
+			request.rocket().state::<Config>().map(|config| config.api_keys().clone())
 		{
 			match keys.len() {
 				0 => Outcome::Failure((Status::Unauthorized, ApiKeyError::Missing)),
-				1 if keys[0] == api_key => Outcome::Success(Guard::default()),
+				1 if allowed_keys.contains(&keys[0].to_string()) => Outcome::Success(ApiKey),
```
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply backend/common/presentation/src/http/guards/api_key.rs:31]: cf la conversation Slack

**Pull request merged**